sql/core/src/main/java/org/apache/spark/sql/sources/v2/ReadSupport.java

```diff
@@ -30,6 +30,9 @@
   /**
    * Creates a {@link DataSourceV2Reader} to scan the data from this data source.
    *
+   * If this method fails (by throwing an exception), the action would fail and no Spark job was
```

Comment:
Are there recommended exceptions to throw? It would be great if this specified what exception classes implementations should use for problems with `@throws` javadoc entries. For example, `@throws AnalysisException If ...`.

Reply:
no exception is recommended here, you can throw any exception and Spark will fail your job. This document just describes what will happen if an exception is thrown, but no exception is expected actually.